src/classes/RD_AddDonationsBTN_TEST.cls

```diff
@@ -143,6 +142,93 @@ private with sharing class RD_AddDonationsBTN_TEST {
         Test.stopTest();
     }
 
+    /***
+    * @description Verify if it displays error message if the user do not have Edit or Delete permissions
+    */
+    @isTest
+    private static void shouldDisplayErrorMessageWhenUserHaveNoEditOrDeletePermission() {
+        String accessType = 'CreateOnly';
+
+        Account acc = UTIL_UnitTestData_TEST.buildOrganizationAccount();
+        insert acc;
+
+        npe03__Recurring_Donation__c rd = getLegacyRDBuilder()
+            .withAccount(acc.Id)
+            .withNextDonationDate(System.today())
+            .build();
+
+        insert rd;
+
+        User readOnlyUser = UTIL_UnitTestData_TEST.createUser(UTIL_UnitTestData_TEST.PROFILE_READONLY_USER);
+
+        User adminUser = UTIL_UnitTestData_TEST.createUserWithoutInsert(UTIL_Profile.SYSTEM_ADMINISTRATOR);
+        // Run as admin user to set the permission set with object level create permission for the readonlyuser
+        System.runas(adminUser) {
+            assignPermissionSet(accessType, readOnlyUser.id);
+        }
+
+
+        Test.startTest();
+
+        System.runAs(readOnlyUser) {
+            RD_AddDonationsBTN_CTRL ctrl = new RD_AddDonationsBTN_CTRL(
+                new ApexPages.StandardController(rd)
+            );
+
+            // The error should be generated and rendered by the constructor
+            UTIL_UnitTestData_TEST.assertPageHasError(System.Label.RD_ErrorAddDonationPermissionDenied);
+            UTIL_UnitTestData_TEST.assertPageHasMessage(System.Label.RD_ErrorAddDonationPermissionDenied, ApexPages.Severity.ERROR);
+            System.assertEquals(false, ctrl.showContinueBtn, 'The continue button should be hidden');
+
+        }
+
+        Test.stopTest();
+
+    }
+
+    /***
+    * @description Verify if it refreshes the opportunity when the read only user has all permissions
+    */
+    @isTest
+    private static void shouldContinueRefreshOpportunityWithAllPermissions() {
```

Comment:
This test might be obsolete since functionality is already tested by `shouldRefreshOppsWhenButtonClick` ie when user has all CRUD on the Opportunity SObject, ie user is an admin, they can refresh the Opportunities.

Tests verifying the following can test various scenarios:
- `shouldDisplayErrorMessageWhenUserHasNoCreatePermission()`
- `shouldDisplayErrorMessageWhenUserHasNoEditPermission()`
- `shouldDisplayErrorMessageWhenUserHasNoDeletePermission()`